### DISCUSSION TITLE
Need help with encrypting an existing database

### DISCUSSION THREAD

**Discussion by srdjanmaksimovic021** [ericsink/SQLitePCL.raw#410] [Q&A]
Hi, first of all thanks for creating and maintaining this awesome package! :)

I am working on a project that uses this. After rolling out a release which locally uses an unencrypted database, we would like to be able to encrypt these user databases in the next release, but do it in a way so that users don't lose existing data.

I found that sqlcypher has a way to do this:
ATTACH DATABASE 'encrypted.db' AS encrypted KEY 'newkey';
SELECT sqlcipher_export('encrypted');
DETACH DATABASE encrypted;

Yet I was unable to make it work and was unable to find any documentation on e_sqlcypher which this package is using. So, basically my question is: Is this possible with your package and e_sqlcypher and if so how would one go about doing that?

Thanks in advance!

**Comment by ericsink** [comment]
Sorry, I can't provide much help with using sqlcipher.  I typically suggest that folks check out the Zetetic website for assistance.
